omc3/scripts/fake_measurement_from_model.py

```diff
@@ -274,32 +279,35 @@ def create_beta(df_twiss: pd.DataFrame, df_model: pd.DataFrame, parameter: str,
 
 
 def create_dispersion(df_twiss: pd.DataFrame, df_model: pd.DataFrame, parameter: str,
-                      relative_error: float, randomize: Sequence[str], headers: dict):
+                      relative_error: float, randomize: Sequence[str], headers: dict,
+                      seed: int | None = None):
     """ Creates dispersion measurement. """
     LOG.info(f"Creating fake dispersion for {parameter}.")
     plane = parameter[-1]
-    df = create_measurement(df_twiss, parameter, relative_error, randomize)
+    df = create_measurement(df_twiss, parameter, relative_error, randomize, seed)
     df = append_model_param(df, df_model, parameter)
     df = append_model_s_and_phaseadv(df, df_model, planes=plane)
     df.headers = headers.copy()
     return {f'{DISPERSION_NAME}{plane.lower()}': df}
 
 
 def create_phase(df_twiss: pd.DataFrame, df_model: pd.DataFrame, parameter: str,
-                 relative_error: float, randomize: Sequence[str], headers: dict):
+                 relative_error: float, randomize: Sequence[str], headers: dict,
+                 seed: int | None = None):
     """ Creates both phase advance and total phase measurements. """
     LOG.info(f"Creating fake phases for {parameter}.")
     results = {}
-    dict_adv = create_phase_advance(df_twiss, df_model, parameter, relative_error, randomize, headers)
+    dict_adv = create_phase_advance(df_twiss, df_model, parameter, relative_error, randomize, headers, seed)
     results.update(dict_adv)
 
-    dict_tot = create_total_phase(df_twiss, df_model, parameter, relative_error, randomize, headers)
+    dict_tot = create_total_phase(df_twiss, df_model, parameter, relative_error, randomize, headers, seed)
     results.update(dict_tot)
     return results
 
 
 def create_phase_advance(df_twiss: pd.DataFrame, df_model: pd.DataFrame, parameter: str,
-                         relative_error: float, randomize: Sequence[str], headers: dict):
+                         relative_error: float, randomize: Sequence[str], headers: dict,
+                         seed: int | None = None):
```

Comment:
This now takes a `seed` and uses it in for its own `Generator`. It passes to `_get_random_errors` if errors are to be randomized too.